### ISSUE TITLE
[Bug]: Invalid Provider Server Combination

### ISSUE THREAD

**Issue by MichaelVoelkel** [hetznercloud/terraform-provider-hcloud#763]
### What happened?

So I basically try to set up a simple Terraform script. Just a token, just a server, really simple (I hope). After the upgrade to 1.44.0 I get:
```
 Error: Failed to load plugin schemas
│ 
│ Error while loading schemas for plugin components: Failed to obtain provider schema: Could not load the schema for provider registry.terraform.io/hetznercloud/hcloud:
│ failed to retrieve schema from provider "registry.terraform.io/hetznercloud/hcloud": Invalid Provider Server Combination: The combined provider has differing provider
│ schema implementations across providers. Provider schemas must be identical across providers. This is always an issue in the provider implementation and should be
│ reported to the provider developers.
```

### What did you expect to happen?

That it gives no error.

### Please provide a minimal working example

So my main tf is:
```
provider "hcloud" {
    token = var.hcloud_token
}

resource "hcloud_ssh_key" "k8s_admin_key" {
  name       = "k8s_admin_key"
  public_key = file(var.ssh_public_key_filepath)
}

resource "hcloud_network" "network" {
    name = "kubenet"
    ip_range = "10.20.0.0/16"
}

resource "hcloud_server" "master" {
    count = var.master_node_count
    name = "${var.cluster_name}-master-${count.index + 1}"
    location = var.location
    server_type = var.master_type
    image = var.node_image
    ssh_keys = [hcloud_ssh_key.k8s_admin_key.id]

    connection {
        host = self.ipv4_address
        type = "ssh"
        private_ssh_key = file(var.private_ssh_key)
    }
}
```

Variables are filled accordingly, I hope all is speaking enough. But I might even have errors here but it does not come to this.

I am doing a fresh init & apply.

**Comment by annismckenzie** [comment]
We also ran into this just now in one of our CI pipelines. I looked at #749 and that refactoring effort breaks the provider.

The provider schema definitions are different:

https://github.com/hetznercloud/terraform-provider-hcloud/pull/749/files#diff-86978c2f5d893139f24fd19d3849c2fcf3f88ddc7cce5190f4e7e0bad2fc0dc7R44-R48

vs.

https://github.com/hetznercloud/terraform-provider-hcloud/pull/749/files#diff-60d4f1b041da05453d0b04458b428b1c0cb4a0108a619d8890165f97facf9177L46-L59

The old provider has `Required: true` and `Optional: false` while the new one has it the other way around. Version v1.43.0 still works.
